### PR TITLE
0.2.0 release

### LINKED ISSUE #85: IconButton
Screen Shot 2021-05-13 at 9 31 12 PM
Screen Shot 2021-05-13 at 9 31 21 PM

### LINKED ISSUE #97: BackButton
Screen Shot 2021-05-18 at 9 25 57 PM

Goes on Titlebars, I think. @lesliearkorful

### LINKED ISSUE #103: Add VisualDensity to MacosThemeData

### LINKED ISSUE #104: Actually return localizations in MaterialApp
We define the as properties but they aren't returned.

### PR DESCRIPTION
**Pull request by GroovinChip**
This PR contains a number of additions and fixes targeting the 0.2.0 milestone:
Closes #85 , #97 , #103 , and #104 

~~Once merged, #81 will need to be addressed in order to merge 0.2.0 to stable.~~

## Pre-launch Checklist

- [x] I have run `dartfmt` on all changed files
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could
- [x] I have added/updated relevant documentation
- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes
- [x] I have run `flutter pub publish --dry-run` and addressed any warnings

### CODE REVIEW

File: lib/src/buttons/icon_button.dart
Comment (bdlukaa) [inline lib/src/buttons/icon_button.dart]: const

File: lib/src/buttons/icon_button.dart
Comment (bdlukaa) [inline lib/src/buttons/icon_button.dart]: delete if not used

File: lib/src/macos_app.dart
Comment (bdlukaa) [inline lib/src/macos_app.dart:314]: `MacosLocalizations` and `DefaultMacosLocalizations` should be created. The back button should have it as its semantic label

File: lib/src/macos_app.dart
Comment (bdlukaa) [inline lib/src/macos_app.dart:318]: add `yield DefaultWidgetsLocalizations.delegate`

File: lib/src/theme/macos_theme.dart
Comment (bdlukaa) [inline lib/src/theme/macos_theme.dart:270]: this line is useless because it's already being defined on line 228

File: lib/src/theme/macos_theme.dart
Comment (bdlukaa) [inline lib/src/theme/macos_theme.dart:277]: so this is

File: lib/src/theme/macos_theme.dart
Comment (bdlukaa) [inline lib/src/theme/macos_theme.dart]: Replace this whole documentation block (from `@template` to `@endTemplate`) to `{@macro flutter.material.themedata.visualDensity}`

File: lib/src/util.dart
Comment (bdlukaa) [inline lib/src/util.dart:37]: These helper methods shouldn't be exported

**Review by bdlukaa — DISMISSED** [review]
I think it should be version 0.1.4 instead of 0.2.0

**Comment by GroovinChip** [comment]
> I think it should be version 0.1.4 instead of 0.2.0

The reason for 0.2.0 is because it involves more than one or two changes. Bigger groups of changes warrant a 0.X.0 bump, small changes get a 0.0.X bump.

**Review by lesliearkorful — DISMISSED** [review]
#106 should land and be merged into this before this is merged into `dev`

File: lib/src/theme/macos_theme.dart
Comment (GroovinChip) [reply lib/src/theme/macos_theme.dart:270]: It's not, actually. This is a different scope.

File: lib/src/theme/macos_theme.dart
Comment (GroovinChip) [reply lib/src/theme/macos_theme.dart:277]: Different scope

**Review by GroovinChip — COMMENTED** [review]
#106 landed and merged

File: lib/src/macos_app.dart
Comment (GroovinChip) [reply lib/src/macos_app.dart:314]: Not sure what mac-specific localizations are needed. We can work that in later, current localizations seem to work fine

**Review by bdlukaa — APPROVED** [review]
LGTM

**Comment by GroovinChip** [comment]
Waiting on #102 to land before merging

**Comment by GroovinChip** [comment]
@lesliearkorful any final thoughts?

**Comment by lesliearkorful** [comment]
LGTM

**Pull request merged**
